cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2401,12 +2413,19 @@ String getCertificateThumbprint(Secret certSecret, CertAndKeySecretSource custom
 
         ConfigMap getKafkaAncillaryCm()    {
             ConfigMap loggingCm = null;
+            ConfigMap metricsCm = null;
 
             if (kafkaCluster.getLogging() instanceof ExternalLogging) {
                 loggingCm = configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) kafkaCluster.getLogging()).getName());
             }
 
-            ConfigMap brokerCm = kafkaCluster.generateAncillaryConfigMap(loggingCm, kafkaAdvertisedHostnames, kafkaAdvertisedPorts);
+            if (kafkaCluster.isMetricsConfigured()) {
+                if (kafkaCluster.getMetricsConfigInCm() instanceof JmxPrometheusExporterMetrics) {
```

Comment:
I'm seeing this repeated enough times to think we should have a method: `kafkaClienter.isMetricsJmxPrometheusExporter()`

Reply:
I left it separated on purpose. Let's say we will support another type of metrics so the code will look like 
```
            if (kafkaCluster.isMetricsConfigured()) {
                if (kafkaCluster.getMetricsConfigInCm() instanceof JmxPrometheusExporterMetrics) {
                   doJmxStuff() 
                }
                 if (kafkaCluster.getMetricsConfigInCm() instanceof AnyDifferrentMetrics) {
                   doDifferentStuff() 
                }
             }
```